### ISSUE TITLE
build fails with hdf5 1.14.0

### ISSUE THREAD

**Issue by ltalirz** [qcserenity/serenity#10]
Specifically, `call of overloaded 'H5File(const char*, unsigned int, int)' is ambiguous`

```
  >> 126    /tmp/awvwgk/spack-stage/spack-stage-serenity-1.4.0-dpss67esoc3i37ah5fuq35bodakiaujk/spack-src/src/data/ElectronicStructure.cpp:198:62: error: call of overloaded 'H5File(const char*, unsigned int, int)' is ambiguous
     127      198 |   HDF5::H5File file(name.c_str(), H5F_ACC_RDONLY, H5P_DEFAULT);
     128          |                                                              ^
     129    In file included from /home/awvwgk/projects/spack/gh/opt/spack/linux-ubuntu20.04-zen2/gcc-10.3.0/hdf5-1.14.0-ah57qce2zxrmcmdrhfuinoorkx3kv2zc/include/H5Cpp.h:47
```

Was fixed by @awvwgk by limiting version range to 1.10.1 - 1.12.2

**Comment by nabbelbabbel** [comment]
Thanks for the report I believe this particular error can be fixed by removing the third argument in that specific call.

I will stage this for the next release.

**Comment by nabbelbabbel** [comment]
Staged for v1.5.0

**Comment by nabbelbabbel** [comment]
This issue has been addressed in Serenity v1.5.0 .
Thanks for the feedback.
